ambry-store/src/test/java/com.github.ambry.store/IndexValueTest.java

```diff
@@ -321,44 +316,24 @@ static IndexValue getIndexValue(IndexValue value, short version) {
   // Instantiation of {@link IndexValue} in version {@link PersistentIndex#VERSION_0}
 
   /**
-   * Constructs IndexValue based on the args passed in, in the version specified
+   * Constructs IndexValue based on the args passed in, in the {@link PersistentIndex#VERSION_0}
    * @param size the size of the blob that this index value refers to
    * @param offset the {@link Offset} in the {@link Log} where the blob that this index value refers to resides
    * @param expiresAtMs the expiration time in ms at which the blob expires
    * @param originalMessageOffset the original message offset where the Put record pertaining to a delete record exists
    *                              in the same log segment. Set to -1 otherwise.
-   * @param version the version of IndexValue which is to be constructed
    * @return the {@link IndexValue} thus constructed
    */
   private static IndexValue getIndexValue(long size, Offset offset, byte flags, long expiresAtMs,
-      long originalMessageOffset, short version) {
-    IndexValue indexValue = null;
-    switch (version) {
-      case PersistentIndex.VERSION_0:
-        ByteBuffer value = ByteBuffer.allocate(IndexValue.INDEX_VALUE_SIZE_IN_BYTES_V0);
-        value.putLong(size);
-        value.putLong(offset.getOffset());
-        value.put(flags);
-        value.putLong(expiresAtMs);
-        value.putLong(originalMessageOffset);
-        value.position(0);
-        indexValue = new IndexValue(offset.getName(), value, PersistentIndex.VERSION_0);
-        break;
-      case PersistentIndex.VERSION_1:
-        value = ByteBuffer.allocate(IndexValue.INDEX_VALUE_SIZE_IN_BYTES_V1);
```

Comment:
We don't manipulate/validate any negative expiration values while initializing IndexValue. So, special serialization code (this code block) is not required. We go via usual path of passing in negative exp value and ser -> deser and check if the value is set to -1